libc/src/search/lfind.cpp

```diff
@@ -0,0 +1,29 @@
+//===-- Implementation of lfind   -------------------------------*- C++ -*-===//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+
+#include "src/search/lfind.h"
+#include "src/__support/CPP/cstddef.h" // cpp::byte
+#include "src/__support/common.h"
+#include "src/__support/macros/config.h"
+
+namespace LIBC_NAMESPACE_DECL {
+LLVM_LIBC_FUNCTION(void *, lfind,
+                   (void *key, void *base, size_t *nmemb, size_t size,
+                    int (*compar)(void *, void *))) {
```

Comment:
Does POSIX say anything about these parameters being `nullptr`? Perhaps we should check the 4 pointer members first before attempting to dereference any of them?

Reply:
The [standard](https://pubs.opengroup.org/onlinepubs/9799919799/functions/lfind.html) does not explicitly mentioned this case. However it does specify the behavior of the function if an "error" occurs.
> Both functions shall return a null pointer in case of error.

However it does not specify what constitutes an error.

I think we could consider this case an "error", return early (with null pointer like the standard says), and that way we avoid the undefined behavior of dereferencing null pointer when it is given as an argument.

Reply:
Resolved by [7cca8d8](https://github.com/llvm/llvm-project/pull/114692/commits/7cca8d80a68c2b0aa83eb21a0b7c8716ee09f284)